### PR TITLE
Fixed the Like feature user lose track of the post

### LINKED ISSUE #1636: Non-logged in user -> like post -> sign In -> lands on FP home page and user lose track of respective post
1. Open "https://staging.fightpandemics.work/"
2. Navigate to Help Board -> Click on 'Like' button for any Post

You will observe that user is directed to Sign In page as expected : PASS
But, user lost the post/track, for which POST user want to LIKE it. 

Please note this issue exist in PROD as well.
Also, if non logged in user attempt to add comment, then user is redirected to sign in and after sign in lands to add on comment page for the same post. so user does not lose track while adding comment as non logged in user but for LIKE feature user lose track of the post after login in.

https://media.giphy.com/media/QQWLHu3QYHOlgcydEw/giphy.gif

### PR DESCRIPTION
**Pull request by sujatha-m**
### What does this pull request aim to achieve? 💯

_Please be concise and link any related issue(s):_This PR Resolves #1636 

- [x] I have checked that no one else is working on similar changes.
- [x] I have read the latest [**README**](README.md) and understand the development workflow.
- [x] The name of this branch is: **`feature/<branch_name>`**, or **`hotfix/<branch_name>`** if this is a hotfix, where `<branch_name>` briefly describes the issue(s) resolved and is prefixed with the issue number(s) (e.g. `1127-1130-update-git-branching-model`). The branch is created in a cloned version of this repo, **not a forked repo**.
- [x] This branch is rebased/merged with the **latest staging** (or the **latest production** if it is a hotfix).
- [x] There are no merge conflicts.
- [x] There are no console warnings and errors.
- [x] On the right hand side of this PR, I have linked the appropriate issue(s) under "Issues" and added a project progress under "Projects".
- [x] The title describes the issue(s) being addressed using format: **`<issue_numbers> - <brief_issue_description>`** (e.g. `#1127/#1130 - Update Git Branching Model`)

### CODE REVIEW

**Comment by nivetheatha** [comment]
When Non logged user lands on helpboard.The like button looks like a comment button instead of heart shaped button.
Screenshot_20201013-160705_Chrome

**Comment by sujatha-m** [comment]
> When Non logged user lands on helpboard.The like button looks like a comment button instead of heart shaped button.
> Screenshot_20201013-160705_Chrome

**Comment by sujatha-m** [comment]
> When Non logged user lands on helpboard.The like button looks like a comment button instead of heart shaped button.
> Screenshot_20201013-160705_Chrome

Fixed

**Comment by nivetheatha** [comment]
Non logged user -->help board-->tries to like the post-->sign in using Facebook -->lands on particular post-->click back-->landed on home page --> click on help board
1.Now the like button is again looks like comment button. 
2.Automatically logged out the user.
Screenshot_20201013-220945_Chrome

**Comment by sujatha-m** [comment]
Tried the same reported sequence,but do not see the issue.Please see the screenshot below.
Did you test with the latest fix?

Screenshot (124)

**Comment by nivetheatha** [comment]
@sujatha-m yes I tested in latest build only.FYI I tested in Android Mobile.
I checked again still its working the same way as mentioned above. If we sign in through email then it works as expected. But if we sign in through Facebook . Then the issue pops up.

**Comment by nivetheatha** [comment]
ezgif-1-bbd61539a35a

**Comment by nivetheatha** [comment]
> Tried the same reported sequence,but do not see the issue.Please see the screenshot below.
> Did you test with the latest fix?
> 
> Screenshot (124)

It works well in desktop..Not working in Mobile only.

**Comment by rabbi08** [comment]
@sujatha-m 
Below is GIF from @nivetheatha for the issue with 'Like' button

GIF-201014_194437

**Comment by nivetheatha** [comment]
0_GIF-201015_124248
Now it works as expected.

**Review by joshmorel — APPROVED** [review]
LGMT Thanks @sujatha-m

File: client/src/components/Feed/PostSocial.js [hunk 1]
```diff
@@ -126,14 +126,44 @@ const PostSocial = ({
...
+            >
+              <div
+                id={gtmTag("like", GTM.feed.prefix)}
+                className="social-icon"
+                onClick={() => handlePostLike(id, liked, true)}
+              >
+                {renderLikeIcon()}
+                {renderLabels("Like", numLikes)}
+              </div>
+            </Link>
```
Comment (Zaydme) [inline client/src/components/Feed/PostSocial.js]: authenticated user should be able to like directly without the need to redirect them to the post page.

```suggestion
            <div
              id={gtmTag("like", GTM.feed.prefix)}
              className="social-icon"
              onClick={() => handlePostLike(id, liked, true)}
            >
              {renderLikeIcon()}
              {renderLabels("Like", numLikes)}
            </div>
```

**Review by Zaydme — CHANGES_REQUESTED** [review]

File: client/src/components/Feed/PostSocial.js (hunk 1 above)
Comment (joshmorel) [reply client/src/components/Feed/PostSocial.js]: Good catch thanks @Zaydme

**Review by Zaydme — APPROVED** [review]

**Pull request merged**